Frontend/CO.CDP.OrganisationApp.Tests/Pages/Registration/OrganisationDetailsSummaryModelTest.cs

```diff
@@ -62,6 +63,216 @@ public async Task OnPost_OnSuccess_RedirectsToOrganisationOverview()
             .Which.PageName.Should().Be("OrganisationOverview");
     }
 
+    [Fact]
+    public async Task OnPost_DuplicateOrganisationName_AddsModelError()
+    {
+        var problemDetails = new OrganisationWebApiClient.ProblemDetails(
+            title: "Duplicate organisation",
+            detail: "An organisation with this name already exists.",
+            status: 400,
+            instance: null,
+            type: null
+        )
+        {
+            AdditionalProperties = new Dictionary<string, object>
+            {
+                { "code", ErrorCodes.ORGANISATION_ALREADY_EXISTS }
+            }
+        };
+
+        var aex = new ApiException<OrganisationWebApiClient.ProblemDetails>(
+            "Duplicate organisation",
+            400,
+            "Bad Request",
+            null,
+            problemDetails,
+            null
+        );
```

Comment:
What would make these tests more readable is removing irrelevant details. A common pattern is to use a factory method to create fixtures required by the test.


For example. if the status code and error code are the only details that impact the test case, we could reduce it to:

```csharp
var problemDetails = GivenProblemDetail(
  code: ErrorCodes.ORGANISATION_ALREADY_EXISTS,
  status: 400
);
var aex = GivenApiExceptionWithProblemDetails(problem: problemDetails);
```

The idea is that `Given*` methods have all the arguments optional (populated with defaults). This way, the details that are relevant to the test case will be visible in it as they'll be passed to the `Given*` method. Irrelevant details will still be populated with defaults and won't impact test readability.

You can see an example of this approach in the `CO.CDP.OrganisationInformation.Persistence` project, where factory methods are extracted to be reused. However, in your case, I'd just define them as private method inside the test case class until they're needed by another test case.

https://github.com/cabinetoffice/GCGS-Central-Digital-Platform/blob/3735d0958d71579a8736e9cf3d7444c5713a84fb/Services/CO.CDP.OrganisationInformation.Persistence.Tests/EntityFactory.cs#L32-L56

https://github.com/cabinetoffice/GCGS-Central-Digital-Platform/blob/3735d0958d71579a8736e9cf3d7444c5713a84fb/Services/CO.CDP.OrganisationInformation.Persistence.Tests/DatabasePersonRepositoryTest.cs#L38-L51

Reply:
Let me also clarify WHY I think it's a good idea to use factories for test fixtures. 

Object construction can get complex as we often need to provide a lot of arguments. This causes a few problems:
* the test become less readable as it mixes details that are relevant to the test with details that are irrelevant (i.e. `code` is relevant as it participates in decision-making, but the `message` is irrelevant as it's ignored).
* there's lots of duplication between tests.
* tests are more prone to cascade changes caused by constructor signature changes.

Reply:
Let me also clarify WHY I think it's a good idea to use factories for test fixtures. 

Object construction can get complex as we often need to provide a lot of arguments. This causes a few problems:
* the test become less readable as it mixes details that are relevant to the test with details that are irrelevant (i.e. `code` is relevant as it participates in decision-making, but the `message` is irrelevant as it's ignored).
* there's lots of duplication between tests.
* tests are more prone to cascade changes caused by constructor signature changes.

Reply:
I understand and agree that using factory methods for creating test fixtures will improve the readability and maintainability of the tests. I will refactor the existing unit tests and extract the common code into a TestEntityFactory class.